addOns/authhelper/src/main/java/org/zaproxy/addon/authhelper/VerificationDetectionProcessor.java

```diff
@@ -136,14 +136,26 @@ private synchronized void updateContext(String loggedInIndicator, String loggedO
         authMethod.setLoggedOutIndicatorPattern(loggedOutIndicator);
         authMethod.setPollData(details.getMsg().getRequestBody().toString());
 
-        String contentType = details.getMsg().getRequestHeader().getHeader(HttpHeader.CONTENT_TYPE);
-        if (contentType != null) {
-            authMethod.setPollHeaders(HttpHeader.CONTENT_TYPE + ": " + contentType);
+        StringBuilder sb = new StringBuilder();
+        appendHeader(sb, details.getMsg().getRequestHeader(), HttpHeader.CONTENT_TYPE);
+        appendHeader(sb, details.getMsg().getRequestHeader(), HttpHeader.REFERER);
+        if (!sb.isEmpty()) {
+            authMethod.setPollHeaders(sb.toString());
         }
         AuthUtils.setVerificationDetailsForContext(context.getId(), details);
         Stats.incCounter("stats.auth.configure.verification");
     }
 
+    private void appendHeader(StringBuilder sb, HttpRequestHeader reqHeader, String headerName) {
+        String headerValue = reqHeader.getHeader(headerName);
+        if (headerValue != null) {
+            sb.append(headerName);
+            sb.append(": ");
+            sb.append(headerValue);
+            sb.append("\n");
+        }
+    }
```

Comment:
I feel like there are quite a few places we do this. Shall I add some convenience methods to core?

Reply:
This is not an HTTP header, it's the configuration of the poll headers, might look similar but not the same (i.e. `\r\n`).